hal/src/nRF52840/sleep_hal.cpp

```diff
@@ -84,13 +95,19 @@ static void unbumpWakeupSourcesPriority(const hal_wakeup_source_base_t* wakeupSo
         } else if (source->type == HAL_WAKEUP_SOURCE_TYPE_BLE) {
             NVIC_SetPriority(SD_EVT_IRQn, priority->blePriority);
             NVIC_DisableIRQ(SD_EVT_IRQn);
+        } else if (source->type == HAL_WAKEUP_SOURCE_TYPE_USART) {
+            NVIC_SetPriority(UARTE0_UART0_IRQn, priority->usart0Priority);
+            nrf_uarte_int_disable(NRF_UARTE0, NRF_UARTE_INT_RXDRDY_MASK);
```

Comment:
We cannot just disable it. The state of RXDRDY should be checked, cached and then restored otherwise I think there might still be a case where the threads blocked waiting a READABLE event will not get it.

Same https://github.com/particle-iot/device-os/pull/2162/files#diff-624879f8eaa372ccb534536b578173daR103